tests/cache/sdkman.bats

```diff
@@ -0,0 +1,17 @@
+#!/usr/bin/env bats
+
+setup() {
+  mkdir -p "${HOME}/cache"
+
+  source /dev/stdin <<< "$(cat ./packages/sdkman.sh)"
+  source $HOME/.sdkman/bin/sdkman-init.sh
+  sdk install grails
+}
+
+teardown() {
+  rm -rf $HOME/.sdkman
+}
+
+@test "[sdkman.sh] Caches Grails package" {
```

Comment:
I would test the following items

* running the `sdkman.sh` script exits with a 0 exit code
* after running the script the `archives` folder is a symlink to `$HOME/cache/sdkman`

I don't think we need to test that `sdkman` can install Grails or another one of their supported tools. They should take care of that on their end. If we do want to add those tests I would do so in a separate test and test the following

* running `sdkman install ??` returns a zero exit code
* the cache folder contains the downloaded archive